src/cs-utils/Signal.hpp

```diff
@@ -92,8 +110,23 @@ class Signal {
   }
 
  private:
+  void postEmitCleanUp() {
+    if (mDisconnectAllRequested) {
+      mSlots.clear();
+      mDisconnectAllRequested = false;
+    } else {
+      for (int id : mSlotsToDisconnect) {
+        mSlots.erase(id);
+      }
```

Comment:
Shouldn't this be cleared? E.g. like this:

```suggestion
      }
      mSlotsToDisconnect.clear();
```